src/main/scala/com/campudus/tableaux/database/model/structure/ColumnModel.scala

```diff
@@ -1,16 +1,137 @@
 package com.campudus.tableaux.database.model.structure
 
 import java.util.NoSuchElementException
+import java.util.concurrent.TimeUnit
 
 import com.campudus.tableaux.database._
 import com.campudus.tableaux.database.domain._
 import com.campudus.tableaux.database.model.TableauxModel._
 import com.campudus.tableaux.helper.ResultChecker._
 import com.campudus.tableaux.{DatabaseException, NotFoundInDatabaseException, ShouldBeUniqueException}
+import com.google.common.cache.CacheBuilder
+import com.google.common.cache.{Cache => GuavaBuiltCache}
 import org.vertx.scala.core.json._
 
 import scala.concurrent.Future
 
+object CachedColumnModel {
+
+  /**
+    * Default never expire
+    */
+  val DEFAULT_EXPIRE_AFTER_ACCESS: Long = -1l
+
+  /**
+    * Max. 10k cached values per column
+    */
+  val DEFAULT_MAXIMUM_SIZE: Long = 10000l
+}
+
+class CachedColumnModel(val config: JsonObject, override val connection: DatabaseConnection)
+  extends ColumnModel(connection) {
+
+  import CachedColumnModel._
+  import scalacache.caching
+  import scalacache.ScalaCache
+  import scalacache.guava.GuavaCache
+
+  implicit val scalaCache = ScalaCache(GuavaCache(createCache()))
+
+  private def createCache(): GuavaBuiltCache[String, Object] = {
+    val builder = CacheBuilder
+      .newBuilder()
+
+    val expireAfterAccess = config.getLong("expireAfterAccess", DEFAULT_EXPIRE_AFTER_ACCESS).toLong
+    if (expireAfterAccess > 0) {
+      builder.expireAfterAccess(expireAfterAccess, TimeUnit.SECONDS)
+    } else {
+      logger.info("Cache will not expire!")
+    }
+
+    val maximumSize = config.getLong("maximumSize", DEFAULT_MAXIMUM_SIZE).toLong
+    if (maximumSize > 0) {
+      builder.maximumSize(maximumSize)
+    }
+
+    builder.recordStats()
+
+    builder.build[String, Object]
+  }
+
+  private def removeCache(tableId: TableId, columnId: Option[ColumnId]): Future[Unit] = {
+    import scalacache.remove
+
+    for {
+      _ <- columnId match {
+        case Some(id) => remove("retrieve", tableId, id)
+        case None => Future.successful(())
+      }
+      _ <- remove("retrieveAll", tableId)
+
+      dependencies <- retrieveDependencies(tableId)
+
+      _ <- Future.sequence(dependencies.map({
+        case (_tableId, _columnId) =>
+          for {
+            _ <- remove("retrieve", _tableId, _columnId)
```

Comment:
Since `tableId` and `columnId` is already in use, maybe `dependencyTableId` and `dependencyColumnId` would be nicer names?
By looking at it, I first thought that `_tableId` was `_.tableId` and an anonymous function...